src/debug/showUserInput.ts

```diff
@@ -0,0 +1,212 @@
+    const element = document.createElement('playwright-touch-indicator');
+    const shadow = element.attachShadow({mode: 'closed'});
+    shadow.innerHTML = `
+      <style>
+          svg {
+              overflow: visible;
+          }
+          circle {
+              fill: black;
+              stroke: white;
+              transition: fill .2s;
+          }
+      </style>
+      <svg width="20" height="20">
+          <circle r="10">
+      </svg>`;
+    element.style.position = 'fixed';
+    element.style.pointerEvents = 'none';
+    element.style.zIndex = String(Number.MAX_SAFE_INTEGER - 1);
+    element.style.left = x + 'px';
+    element.style.top = y + 'px';
+    element.animate([{
+      opacity: 0.75,
+    }, {
+      opacity: 0,
+    }], {
+      duration: 500,
+      direction: 'normal'
+    }).onfinish = () => {
+      element.remove();
+    };
+    document.documentElement.appendChild(element);
+  }, [x, y] as const);
+}
+
+async function showMouseIndicator(page: Page, x: number, y: number, buttons: Set<types.MouseButton>) {
+  const args = [x, y, !!buttons.size] as const;
+  await (await page.mainFrame()._utilityContext()).evaluateInternal(([x, y, pressed]) => {
+    const element: HTMLElement = document.querySelector(':scope > playwright-mouse-indicator') || createMouseIndicator();
+    element.style.left = x + 'px';
+    element.style.top = y + 'px';
+    element.classList.toggle('pressed', pressed);
+    // eslint-disable-next-line no-unused-expressions
+    window.getComputedStyle(element).fill;
+
+
+    function createMouseIndicator() {
+      const element = document.createElement('playwright-mouse-indicator');
+      const shadow = element.attachShadow({mode: 'closed'});
+      shadow.innerHTML = `
+      <style>
+        svg {
+          filter: drop-shadow(0px 2px 2px rgba(0,0,0,0.3));
+          opacity: 0.5;
+        }
+        polygon {
+          fill: white;
+          stroke: black;
+          transition: fill .2s;
+        }
+        :host(.pressed) polygon {
+          transition: none;
+          fill: black;
+          transition: none;
+        }
+      </style>
+      <svg width="40" height="40">
+        <polygon points="0,0 0,34 8,28 14,38 20,35 14,26 24,24"/>
+      </svg>`;
+      element.style.position = 'fixed';
+      element.style.pointerEvents = 'none';
+      element.style.zIndex = String(Number.MAX_SAFE_INTEGER - 1);
+      document.documentElement.appendChild(element);
+      return element;
+    }
+  }, args);
+}
+
+const textForPage = new WeakMap<Page, string>();
+async function updateKeyboardIndicator(page: Page, key: string, code: string, modifiers: Set<types.KeyboardModifier>) {
+  let text;
+  let dim = true;
+  if (!key) {
+    if (!textForPage.get(page))
+      return;
+    textForPage.delete(page);
+    text = '';
+  } else if (!modifiers.size && key.trim().length === 1) {
+    const currentText = (textForPage.get(page) || '') + key;
+    textForPage.set(page, currentText);
```

Comment:
I think we should reset this somewhere, otherwise multiple texts will coalesce. Perhaps on other forms of input? Navigations? Timeout?

More generally, I think the text should increase while the tooltip is shown, and be reset once the tooltip is hidden. wdyt?

```js
await page.fill('username', 'John Doe');
await page.check('remember be');
await page.fill('password', 'foo');
```

Reply:
Yeah I think theres a lot we can do by taking into account the specific method that called into the keyboard. I was looking at a karaoke style https://youtu.be/OU3699R53rs?t=23 for page.type. And we can remove the indicator on page.fill if its filling uninterrupted into a textarea. It's just redundant there.

I'd rather tackle these in a follow up patch because they involve a lot of api changes and bikeshedding.